pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1000,6 +1000,11 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 			reason = reasonStartupProbe
 		} else if isInPlacePodVerticalScalingAllowed(pod) && !m.computePodResizeAction(pod, idx, containerStatus, &changes) {
```

Comment:
Should the backoff be cleared even if the container doesn't require a restart? I think yes. 2 use cases for this I can think of:
1. Runtime decides to restart anyway (since we don't ever guarantee no restart)
2. Container was stuck in backoff due to oom-kills, so the user updates the container memory limit. It would be nice if the effect was immediate.

Reply:
Thanks for the suggestion, I will need to perform some manual tests and if possible adding tests to capture those corner cases. Will ping you back

Reply:
@tallclair  I did some tests with your suggestion for the use cases you mention, i think it will require more effort and to create some tests to be more confident. I prefer to create a new issue for items 1 and 2,  keep this for InPlacePodVerticalScaling ( updated title ) to resolve https://github.com/kubernetes/kubernetes/issues/125843 and possible https://github.com/kubernetes/kubernetes/issues/119838. What do you think?

Reply:
I disagree - why not just address this now? I don't understand your comment about confidence. I don't think it increases the scope much to always restart?

Reply:
Ok, will give it a try, perhaps i misunderstood what was asked and overestimated the effort.

Reply:
I guess you meant "resize restart" PTAL

Reply:
@tallclair i did manual test with scenario 2, works fine with latest patch. Please take another look and let me know. if this is what you meant. Proof below with screencast

https://github.com/user-attachments/assets/8bb38150-c1ac-4574-98ea-3013eb758007

With regard #119838 restart counts increase but container in my lab doesn't return back after being patched with more memory. So i don't think this patch "fully" solves the issue ( will update description )

Reply:
Is above "proof" @tallclair @SergeyKanzhelev  sufficient for this commit to be approved or we will need the creation of tests now ?